File: common/src/main/java/DahRealPanda/plugins/colonyweb/platform/ItemSlots.java
package DahRealPanda.plugins.colonyweb.platform;

import net.minecraft.world.item.ItemStack;

/**
 * A read-only view of an inventory, whatever the loader calls its item handler.
 *
 * <p>Forge and NeoForge both have an {@code IItemHandler} with exactly these two methods, in
 * different packages. Shared code talks to this instead so it does not have to name either.</p>
 */
public interface ItemSlots {

    int getSlots();

    ItemStack getStackInSlot(int slot);
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/platform/ItemSlots.kt
package DahRealPanda.plugins.colonyweb.platform

import net.minecraft.world.item.ItemStack

/**
 * A read-only view of an inventory, whatever the loader calls its item handler.
 *
 * Forge and NeoForge both have an [IItemHandler] with exactly these two methods, in
 * different packages. Shared code talks to this instead so it does not have to name either.
 */
interface ItemSlots {
    fun getSlots(): Int
    fun getStackInSlot(slot: Int): ItemStack
}
